diff --git a/node_modules/pkg/lib-es5/packer.js b/node_modules/pkg/lib-es5/packer.js
index 7295bb6..76805a3 100644
--- a/node_modules/pkg/lib-es5/packer.js
+++ b/node_modules/pkg/lib-es5/packer.js
@@ -128,6 +128,7 @@ function _default({
           const newStat = Object.assign({}, value);
           newStat.isFileValue = value.isFile();
           newStat.isDirectoryValue = value.isDirectory();
+          newStat.isSocketValue = value.isSocket();
           const buffer = Buffer.from(JSON.stringify(newStat));
           stripes.push({
             snap,
diff --git a/node_modules/pkg/prelude/bootstrap.js b/node_modules/pkg/prelude/bootstrap.js
index 0d19f1d..db69015 100644
--- a/node_modules/pkg/prelude/bootstrap.js
+++ b/node_modules/pkg/prelude/bootstrap.js
@@ -925,8 +925,10 @@ function payloadFileSync (pointer) {
 
     var isFileValue = s.isFileValue;
     var isDirectoryValue = s.isDirectoryValue;
+    var isSocketValue = s.isSocketValue;
     delete s.isFileValue;
     delete s.isDirectoryValue;
+    delete s.isSocketValue;
 
     s.isFile = function () {
       return isFileValue;
@@ -934,6 +936,9 @@ function payloadFileSync (pointer) {
     s.isDirectory = function () {
       return isDirectoryValue;
     };
+    s.isSocket = function () {
+      return isSocketValue;
+    };
     s.isSymbolicLink = function () {
       return false;
     };
