diff --git a/node_modules/@oclif/core/lib/cli-ux/list.js b/node_modules/@oclif/core/lib/cli-ux/list.js
index 607d8dc..07ba1f2 100644
--- a/node_modules/@oclif/core/lib/cli-ux/list.js
+++ b/node_modules/@oclif/core/lib/cli-ux/list.js
@@ -22,7 +22,7 @@ function renderList(items) {
         }
         left = left.padEnd(maxLength);
         right = linewrap(maxLength + 2, right);
-        return `${left}  ${right}`;
+        return `${left} : ${right}`;
     });
     return lines.join('\n');
 }
diff --git a/node_modules/@oclif/core/lib/help/command.js b/node_modules/@oclif/core/lib/help/command.js
index 82d9eed..0a8d45f 100644
--- a/node_modules/@oclif/core/lib/help/command.js
+++ b/node_modules/@oclif/core/lib/help/command.js
@@ -48,7 +48,7 @@ class CommandHelp extends formatter_1.HelpFormatter {
         if (args.filter((a) => a.description).length === 0)
             return;
         return args.map((a) => {
-            const name = a.name.toUpperCase();
+            const name = a.required ? `<${a.name}>` : `[${a.name}]`;
             let description = a.description || '';
             if (a.default)
                 description = `${(0, theme_1.colorize)(this.config?.theme?.flagDefaultValue, `[default: ${a.default}]`)} ${description}`;
@@ -143,14 +143,12 @@ class CommandHelp extends formatter_1.HelpFormatter {
             label = labels.join((0, theme_1.colorize)(this.config?.theme?.flagSeparator, flag.char ? ', ' : '  '));
         }
         if (flag.type === 'option') {
-            let value = flag.helpValue || (this.opts.showFlagNameInTitle ? flag.name : '<value>');
+            let value = flag.helpValue || (this.opts.showFlagNameInTitle ? flag.name : `<${flag.name}>`);
             if (!flag.helpValue && flag.options) {
                 value = showOptions || this.opts.showFlagOptionsInTitle ? `${flag.options.join('|')}` : '<option>';
             }
             if (flag.multiple)
-                value += '...';
-            if (!value.includes('|'))
-                value = chalk_1.default.underline(value);
+                value += ' ...';
             label += `=${value}`;
         }
         return label;
diff --git a/node_modules/@oclif/core/lib/help/index.js b/node_modules/@oclif/core/lib/help/index.js
index 242538a..efde8ac 100644
--- a/node_modules/@oclif/core/lib/help/index.js
+++ b/node_modules/@oclif/core/lib/help/index.js
@@ -168,11 +168,12 @@ class Help extends HelpBase {
         }
         this.log(this.formatCommand(command));
         this.log('');
-        if (subTopics.length > 0) {
+        const SUPPRESS_SUBTOPICS = true;
+        if (subTopics.length > 0 && !SUPPRESS_SUBTOPICS) {
             this.log(this.formatTopics(subTopics));
             this.log('');
         }
-        if (subCommands.length > 0) {
+        if (subTopics.length > 0 && !SUPPRESS_SUBTOPICS) {
             const aliases = [];
             const uniqueSubCommands = subCommands.filter((p) => {
                 aliases.push(...p.aliases);
diff --git a/node_modules/@oclif/core/lib/parser/errors.js b/node_modules/@oclif/core/lib/parser/errors.js
index 656ec6b..2bbf36b 100644
--- a/node_modules/@oclif/core/lib/parser/errors.js
+++ b/node_modules/@oclif/core/lib/parser/errors.js
@@ -14,7 +14,8 @@ Object.defineProperty(exports, "CLIError", { enumerable: true, get: function ()
 class CLIParseError extends errors_1.CLIError {
     parse;
     constructor(options) {
-        options.message += '\nSee more help with --help';
+        const help = options.command ? `\`${options.command} --help\`` : '--help';
+        options.message += `\nSee more help with ${help}`;
         super(options.message, { exit: options.exit });
         this.parse = options.parse;
     }
@@ -37,7 +38,8 @@ exports.InvalidArgsSpecError = InvalidArgsSpecError;
 class RequiredArgsError extends CLIParseError {
     args;
     constructor({ args, exit, flagsWithMultiple, parse, }) {
-        let message = `Missing ${args.length} required arg${args.length === 1 ? '' : 's'}`;
+        const command = 'balena ' + parse.input.context.id.replace(/:/g, ' ');
+        let message = `Missing ${args.length} required argument${args.length === 1 ? '' : 's'}`;
         const namedArgs = args.filter((a) => a.name);
         if (namedArgs.length > 0) {
             const list = (0, list_1.renderList)(namedArgs.map((a) => [a.name, a.description]));
@@ -48,7 +50,7 @@ class RequiredArgsError extends CLIParseError {
             message += `\n\nNote: ${flags} allow${flagsWithMultiple.length === 1 ? 's' : ''} multiple values. Because of this you need to provide all arguments before providing ${flagsWithMultiple.length === 1 ? 'that flag' : 'those flags'}.`;
             message += '\nAlternatively, you can use "--" to signify the end of the flags and the beginning of arguments.';
         }
-        super({ exit: cache_1.default.getInstance().get('exitCodes')?.requiredArgs ?? exit, message, parse });
+        super({ exit: cache_1.default.getInstance().get('exitCodes')?.requiredArgs ?? exit, message, parse, command });
         this.args = args;
     }
 }
