diff --git a/node_modules/oclif/lib/commands/pack/win.js b/node_modules/oclif/lib/commands/pack/win.js
index ca99b35..6265d26 100644
--- a/node_modules/oclif/lib/commands/pack/win.js
+++ b/node_modules/oclif/lib/commands/pack/win.js
@@ -89,6 +89,12 @@ InstallDir "\$PROGRAMFILES${arch === 'x64' ? '64' : ''}\\${config.dirname}"
 ${customization}
 
 Section "${config.name} CLI \${VERSION}"
+  ; First remove any old client files.
+  ; (Remnants of old versions were causing CLI errors)
+  ; Initially tried running the Uninstall.exe, but was
+  ; unable to make script wait for completion (despite using _?)
+  DetailPrint "Removing files from previous version."
+  RMDir /r "$INSTDIR\\client"
   SetOutPath $INSTDIR
   File /r bin
   File /r client
diff --git a/node_modules/oclif/lib/tarballs/build.js b/node_modules/oclif/lib/tarballs/build.js
index 90904ec..012c7b9 100644
--- a/node_modules/oclif/lib/tarballs/build.js
+++ b/node_modules/oclif/lib/tarballs/build.js
@@ -189,6 +189,11 @@ const extractCLI = async (tarball, c) => {
         (0, promises_1.rm)(node_path_1.default.join(workspace, node_path_1.default.basename(tarball)), { recursive: true }),
         (0, fs_extra_1.remove)(node_path_1.default.join(workspace, 'bin', 'run.cmd')),
     ]);
+    // The oclif installers are a production installation, while the source
+    // `bin` folder may contain a `.fast-boot.json` file of a dev installation.
+    // This has previously led to issues preventing the CLI from starting, so
+    // delete `.fast-boot.json` (if any) from the destination folder.
+    await (0, fs_extra_1.remove)(node_path_1.default.join(workspace, 'bin', '.fast-boot.json'));
 };
 const buildTarget = async (target, c, options) => {
     if (target.platform === 'win32' && target.arch === 'arm64' && (0, semver_1.lt)(c.nodeVersion, '20.0.0')) {
