diff --git a/node_modules/oclif/lib/commands/pack/macos.js b/node_modules/oclif/lib/commands/pack/macos.js
index d06d0b3..c571fe3 100644
--- a/node_modules/oclif/lib/commands/pack/macos.js
+++ b/node_modules/oclif/lib/commands/pack/macos.js
@@ -177,7 +177,8 @@ class PackMacos extends core_1.Command {
             if (process.env.OSX_KEYCHAIN)
                 args.push('--keychain', process.env.OSX_KEYCHAIN);
             args.push(dist);
-            await exec(`pkgbuild  ${args.join(' ')}`);
+            console.error(`[debug] oclif pkgbuild "${args.join('" "')}"`);
+            await exec(`pkgbuild  "${args.join('" "')}"`);
         };
         const arches = _.uniq(buildConfig.targets
             .filter(t => t.platform === 'darwin')
